### ISSUE TITLE
Make Bootstrap fetch Certs data on demand

### ISSUE THREAD

**Issue by arvindh123** [Mainflux/mainflux-lite#1676]
The BOOTSTRAP service stores the Private and Public key, 
 Instead of Bootstrap service,  CERTS service can store the private and public key 
 The Bootstrap can get the Private key and public key from Certs service via SDK 

At present cert renewal process in CERTS service is depend on BOOTSTRAP service. 
Incase of BOOSTRAP service is unavailable the CERTS service can't able renew the certificate. 

And From my point of view , the BOOTSTRAP service storing certificate seems like its storing things which is not related to BOOTSTRAP, the certificates are belongs to CERTS service. Bootstrap can get certificates from CERTS service. 
This similar process are followed in other service like thing, user ,auth , 

ANTOHER IMPORTANTANCE FOR THIS  PROPOSAL, 
During the auto renewal of certificates,  The  renewed certificate need to update in BOOTSTRAP service

The BOOTSTRAP service required TOKEN to update the certificate of configuration , 

The CERTS service sends certificate update request with JWT Token, Thing ID  and renewed certificate to BOOTSTRAP service via SDK , 
**If you see here, JWT Token owner and Thing ID owner will not be the same**
 

So In the BOOTSTRAP JWT Token owner and Thing ID owner will not match and return 404 result . 

For all the above reasons, we can store the certificate to CERTS service and remove the certificate from BOOTSTRAP service

**Comment by dborovcanin** [comment]
I agree, this flow sounds more suitable for our needs. Especially since we can use it for cert renewal, as well. That is, we can make cert renewal _on-demand_: when bootstrapping, if the cert is expired, we execute the renewal procedure. If we're going to do it like this, the question is: do we need certs service or Boostrap can communicate directly with PKI provider/Vault? @drasko @arvindh123 What do you think?

**Comment by arvindh123** [comment]
As you know , the certs service is kind of mediator which is helps to manage the certificates from PKI provider. 
In addition, user can create ,revoke certificate manually. 

We can add endpoint in bootstrap service to do manual update of certificate , but I feel like it is out of scope of bootstrap .  
From my understanding of bootstrap service, It helps to provisioning the things. I don't think it should manage certificates.

And also we can add a feature  "Enable multiple certificates for a single Thing" to certs service which is  mentioned in issue  #1677  
This feature helps to keep multiple certificate for one thing. 
example : one certificate of a thing can be use to write and another of the same thing can be used read 

@dborovcanin , @drasko

**Comment by arvindh123** [comment]
@dborovcanin @drasko ,
One more thing I need to add we can't retrieve private key from PKI,  So we need a place to store the private key, and at the same place we can store public key, For this reason we can store the both private key and public keys on certs service , So on request of a thing certificate, we can send response by retrieving the  private and public key from certs repo

**Comment by dborovcanin** [comment]
> As you know , the certs service is kind of mediator which is helps to manage the certificates from PKI provider. In addition, user can create ,revoke certificate manually.
> 
> We can add endpoint in bootstrap service to do manual update of certificate , but I feel like it is out of scope of bootstrap . From my understanding of bootstrap service, It helps to provisioning the things. I don't think it should manage certificates.
> 
> And also we can add a feature "Enable multiple certificates for a single Thing" to certs service which is mentioned in issue #1677 This feature helps to keep multiple certificate for one thing. example : one certificate of a thing can be use to write and another of the same thing can be used read
> 
> @dborovcanin , @drasko

Makes sense. I'm not sure if we need a `Enable multiple certificates for a single Thing` at all, or if we can simply always use that option. If you renew the cert - you know which one to replace anyways.

**Comment by arvindh123** [comment]
> > As you know , the certs service is kind of mediator which is helps to manage the certificates from PKI provider. In addition, user can create ,revoke certificate manually.
> > We can add endpoint in bootstrap service to do manual update of certificate , but I feel like it is out of scope of bootstrap . From my understanding of bootstrap service, It helps to provisioning the things. I don't think it should manage certificates.
> > And also we can add a feature "Enable multiple certificates for a single Thing" to certs service which is mentioned in issue #1677 This feature helps to keep multiple certificate for one thing. example : one certificate of a thing can be use to write and another of the same thing can be used read
> > @dborovcanin , @drasko
> 
> Makes sense. I'm not sure if we need a `Enable multiple certificates for a single Thing` at all, or if we can simply always use that option. If you renew the cert - you know which one to replace anyways.

@dborovcanin, 
At present, In certs table, we have primary key on three column thing_id, certificate_serial and owner_id,
We can renew and replace the certificate but on renewal process from vault we get new certificate with same common name, the serial number  in renewed (new) certificate will have new serial number. So I think, keeping certificate serial in primary key doesn't help us.
